odata-json-format/10 Media Entity.md

```diff
@@ -33,19 +33,24 @@ An individual property or operation response is represented as a JSON
 object.
 
 A single-valued property or operation response that has the
-`null` value does not have a representation; see
-[OData-Protocol](#ODataProtocol).
+`null` value does not have a representation provided that it carries no
+[control information](#ControlInformation) other than
+[`context`](#ControlInformationcontextodatacontext)
+and no [instance annotations](#InstanceAnnotations); see [OData-Protocol](#ODataProtocol).
 
 A property or operation response that is of a primitive type is
-represented as an object with a single name/value pair, whose name is
+represented as an object with a name/value pair whose name is
 `value` and whose value is a [primitive
-value](#PrimitiveValue).
+value](#PrimitiveValue) or `null`.
 
 A property or operation response that is of complex type is represented
-as a [complex value](#ComplexValue).
+as a [complex value](#ComplexValue). If the value is `null`, the JSON object
```

Comment:
Are entity-valued results covered by this paragraph on complex values as well? They are not covered elsewhere.

Reply:
How do I distinguish `null` from a complex instance without properties?

Reply:
> Entity-valued responses are covered in [6 Entity](https://docs.oasis-open.org/odata/odata-json-format/v4.02/csd01/odata-json-format-v4.02-csd01.html#Entity)

This does not mention entities whose value is `null` (a nullable single-valued navigation property, for example).

Reply:
> This does not mention entities whose value is `null` (a nullable single-valued navigation property, for example).

That is in [8.3 Expanded Navigation Property](https://docs.oasis-open.org/odata/odata-json-format/v4.02/csd01/odata-json-format-v4.02-csd01.html#ExpandedNavigationProperty):

> If at most one entity can be related, the value is the representation of the related entity, or `null` if no entity is currently related.

Reply:
I wasn't talking about _expansion_ but about requesting a _related_ entity, which is absent, like `GET Employees('CEO')/Manager`. I would expect 204 No Content here (unless there is control information or instance annotations).

Or about invoking an operation with entity-valued return type, if that returns null.

Reply:
> requesting a _related_ entity, which is absent, like `GET Employees('CEO')/Manager`

That is covered in [11.2.7 Requesting Related Entities](https://docs.oasis-open.org/odata/odata/v4.02/csd01/part1-protocol/odata-v4.02-csd01-part1-protocol.html#RequestingRelatedEntities):

> If the relationship terminates on a single entity, the response MUST be the format-specific representation of the related single entity. If no entity is related, the service returns [204 No Content](https://docs.oasis-open.org/odata/odata/v4.02/csd01/part1-protocol/odata-v4.02-csd01-part1-protocol.html#ResponseCode204NoContent).

"No Content" means no representation, so this is not mentioned in the JSON Format.